experimental/rocm/dynamic_symbol_tables.h

```diff
@@ -0,0 +1,59 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+RC_PFN_DECL(hipCtxCreate, hipCtx_t *, unsigned int, hipDevice_t)
+RC_PFN_DECL(hipCtxDestroy, hipCtx_t)
+RC_PFN_DECL(hipDeviceGet, hipDevice_t *, int)  // No direct, need to modify
+RC_PFN_DECL(hipGetDeviceCount, int *)
+RC_PFN_DECL(hipDeviceGetName, char *, int, hipDevice_t)  // No direct, need to modify
+RC_PFN_STR_DECL(hipGetErrorName,
+                hipError_t)  // hipGetErrorName(hipError_t) but return const char* instead of hipError_t
+RC_PFN_STR_DECL(hipGetErrorString,
+                hipError_t)  // hipGetErrorName(hipError_t) but return const char* instead of hipError_t
```

Comment:
I'm not sure I understand this comment

Reply:
hipGetErrorString and hipGetErrorName returns char* instead of hipError/CUresult like all the other functions, so I added a different macro to handle that